### PR TITLE
[Mgmt Generator] Fix ParameterContextRegistry duplicate-name crash

### LINKED ISSUE #58484: [Mgmt Generator] ParameterContextRegistry.FindParameter crashes when multiple parameters share the same WireInfo.SerializedName
### Bug Report: MPG Generator crashes in ParameterContextRegistry.FindParameter when multiple parameters share the same WireInfo.SerializedName

**Generator Version**: `@azure-typespec/http-client-csharp-mgmt` v1.0.0-alpha.20260419.1

**Crash Location**: `eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/Models/OperationContexts/ParameterContextRegistry.cs` line 149

**Error**: `System.InvalidOperationException: Sequence contains more than one matching element` at `SingleOrDefault`

---

### Root Cause

`ParameterContextRegistry.FindParameter` at line 149 uses:
```csharp
SingleOrDefault(p => p.WireInfo.SerializedName == parameterToFind.WireInfo.SerializedName)
```
This crashes when two or more parameters in `methodParameters` share the same `WireInfo.SerializedName`. Two distinct scenarios trigger this:

---

### Scenario 1: Duplicate `@segment("providers")` path parameters

**Spec pattern** (`routes.tsp`):
```typespec
@autoRoute @get @action("balances")
getByBillingAccount(
    @path @segment("providers") @key providerNamespace: "Microsoft.Billing",
    @path @segment("billingAccounts") billingAccountId: string,
    ...Azure.ResourceManager.Legacy.Provider,
): Balance | ErrorResponse;
```

**Problem**: The explicit `@segment("providers")` on `providerNamespace` and the spread `...Azure.ResourceManager.Legacy.Provider` (which adds another `@path @segment("providers") provider`) both produce path parameters with `WireInfo.SerializedName = "providers"`. When `FindParameter` searches by serialized name, `SingleOrDefault` finds 2 matches and throws.

**Affected operations**: Any operation that references an external resource provider (e.g., Microsoft.Billing, Microsoft.Capacity, Microsoft.Management) via `@segment("providers")` while also using `...Legacy.Provider` for Microsoft.Consumption. In the Consumption spec, **16 operations** are affected.

**Workaround**: Replace `@autoRoute + @action + @segment + ...Legacy.Provider` with an explicit `@route(...)` path. However, this changes the generated swagger/openapi.json output, which may not be acceptable for spec compliance.

---

### Scenario 2: `scope` query parameter name conflicts with auto-created `scopeParameter`

**Spec pattern** (`routes.tsp`):
```typespec
@autoRoute @get @action("reservationRecommendationDetails")
get(
    @path(#{ allowReserved: true }) @segment("") resourceScope: string,
    ...Azure.ResourceManager.Legacy.Provider,
    @query("scope") scope: Scope,
    ...
): ReservationRecommendationDetailsModel;
```

**Problem**: `MockableArmClientProvider.BuildMethods()` (line 67) creates a `scopeParameter = new ParameterProvider("scope", ...)` for ALL extension-scoped non-resource methods. The `ParameterProvider` constructor auto-generates `WireInfo` from the parameter name when none is explicitly provided, so `new ParameterProvider("scope", ...)` produces `WireInfo.SerializedName = "scope"`. The TypeSpec query parameter `@query("scope")` also produces `WireInfo.SerializedName = "scope"`. Both end up in `methodParameters`, causing `SingleOrDefault` to find 2 matches.

**Key detail**: Simply renaming the TypeSpec parameter name (e.g., `reservationScope`) while keeping `@query("scope")` does NOT fix the issue, because `WireInfo.SerializedName` comes from the `@query` decorator value, not the parameter name.

**Workaround**: Change `@query("scope")` to `@query("reservationScope")` in TypeSpec. But this changes the HTTP wire format from `?scope=Single` to `?reservationScope=Single`, which breaks API compatibility with the actual Azure service endpoint.

---

### Suggested Fix

In `ParameterContextRegistry.FindParameter`, consider:
1. Using `FirstOrDefault` instead of `SingleOrDefault` when a unique match isn't guaranteed
2. Adding a secondary discriminator (e.g., parameter location: path vs query vs header) when multiple parameters share the same serialized name
3. For the scope conflict specifically: the auto-created `scopeParameter` in `MockableArmClientProvider` should use a distinct `WireInfo.SerializedName` (or skip WireInfo entirely since it's a synthetic ResourceIdentifier parameter, not a wire-transmitted parameter)

### Reproduction

Any TypeSpec spec with these patterns will crash the C# MPG generator. The `Azure.ResourceManager.Consumption` spec (api-version 2024-08-01) reproduces both scenarios.

### Impact

- **Consumption SDK migration** is blocked from clean regeneration
- Any future spec using `@segment("providers")` with `...Legacy.Provider`, or having a `@query("scope")` on extension-scoped operations, will hit the same crash

### PR DESCRIPTION
**Pull request by live1206**
## Summary
- fix ParameterContextRegistry so duplicate WireInfo.SerializedName values no longer crash generation
- add regression tests for the duplicate providers and scope collision scenarios
- mark the synthetic ArmClient scope parameter as non-wire so it does not interfere with real request parameters
- migrate test asserts in `Azure.Generator.Management.Tests.Common` (`BinaryDataAssert`, `ModelTests`) from NUnit 3 classic API to the NUnit 4 constraint-based API to fix build errors

## Validation
- dotnet test .\eng\packages\http-client-csharp-mgmt\generator\Azure.Generator.Management\test\Azure.Generator.Mgmt.Tests.csproj --no-restore
- .\eng\scripts\Generate.ps1 -filter 'Mgmt-TypeSpec' -LaunchOnly $true
- .\eng\scripts\Generate.ps1 -filter 'Mgmt-TypeSpec-MultiService' -LaunchOnly $true

Fixes #58484

### CODE REVIEW

File: eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/Models/OperationContexts/ParameterContextRegistry.cs [hunk 1]
```diff
@@ -155,6 +155,65 @@ static ValueExpression FindParameter(IReadOnlyList<ParameterProvider> parameters
             return Default;
         }
 
+        static ParameterProvider? FindBestParameterMatch(IReadOnlyList<ParameterProvider> parameters, ParameterProvider parameterToFind)
+        {
+            var serializedName = parameterToFind.WireInfo?.SerializedName;
```
Comment (ArthurMa1978) [inline eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/Models/OperationContexts/ParameterContextRegistry.cs]: I think the emitter should throw dup parameter error

Comment (live1206) [reply eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/Models/OperationContexts/ParameterContextRegistry.cs]: This is actually not needed.

**Review by ArcturusZhang — APPROVED** [review]
